src/main.rs

```diff
@@ -209,18 +239,42 @@ where
         unreachable!();
     };
 
+    let bench_samples = if let Subcommand::Bench(BenchCmd { bench_samples, .. }) = subcmd {
+        // We need an exact number of samples when benchmarking.
+        match *bench_samples {
+            // The user specified a number of samples.
+            Some(bench_samples) => Some(bench_samples),
+            // The user didn't specify a number of samples, so we estimate
+            // a good number of samples.
+            None => Some(synthesized_impl.estimate_optimal_itrs()?),
+        }
+    } else {
+        None
+    };
+
     match args.format {
         OutputFormat::C => {
-            synthesized_impl.emit(&mut ToWriteFmt(io::stdout()))?;
+            synthesized_impl.emit(bench_samples, &mut ToWriteFmt(io::stdout()))?;
         }
         OutputFormat::Impl => {
             // TODO: How to use Compact? Should we?
             pprint(synthesized_impl, args.impl_style)
         }
     }
-    if args.run {
-        let output = synthesized_impl.build()?.run()?;
-        println!("\nOutput:\n{}", String::from_utf8_lossy(&output.stdout));
+
+    match subcmd {
+        Subcommand::Run(_) => {
+            let output = synthesized_impl.build()?.run()?;
+            println!("\nOutput:\n{}", String::from_utf8_lossy(&output.stdout));
+        }
+        Subcommand::Bench(BenchCmd { .. }) => {
+            let result = synthesized_impl.bench(
+                bench_samples.unwrap(), /* TODO: We know this is not None */
```

Comment:
What is the action item? Not sure what this comment means.

Reply:
@samkaufman Since we are in the `Bench` branch, we know `bench_samples` is not `None`. But `.unwrap()` is safe, so this code is not optimal. We can also use `.unwrap_unchecked()`, but I was not sure if we want to do this with `unsafe` block or if we can come up with another idea. If you think the overhead of `.unwrap()` is negligible, I will remove this comment.

Reply:
Got it. Yes, this is negligible, so let's drop the comment.

Reply:
Sure!